### PR TITLE
Bumping Elysia to 1.0.7 and fixing InferContext

### PR DESCRIPTION
**Pull request by MariNatsuki**
Bumping Elysia to 1.0.7 and fixing InferContext to be compatible with the new Elysia version, addressing issue #14

### CODE REVIEW

File: package.json
Comment (bogeychan) [inline package.json:1]: If bumping elysia contains a breaking change (like `1.0.0` -> `1.0.7`), please also update the `peerDependency` here: https://github.com/bogeychan/elysia-logger/blob/6d0c53d617a541994f6bed87cfb7463d02ff6077/package.json#L35-L37

File: src/types.ts [hunk 1]
```diff
@@ -76,17 +70,24 @@ interface BaseLoggerOptions extends pino.LoggerOptions {}
...
-  infer _EphemeralMetadata
+  infer Metadata,
+  infer _Routes,
+  infer Ephemeral,
+  infer Volatile
 >
-  ? Context<Routes, SingletonBase, Path> &
-      Partial<Singleton["derive"]> &
-      Partial<Singleton["decorator"]>
+  ? Partial<
```
Comment (bogeychan) [inline src/types.ts:82]: Why do you `Partial` the whole `Context` over here?

File: src/types.ts [hunk 2]
```diff
@@ -76,17 +70,24 @@ interface BaseLoggerOptions extends pino.LoggerOptions {}
...
+      Context<
+        Metadata["schema"] & Ephemeral["schema"] & Volatile["schema"],
```
Comment (bogeychan) [inline src/types.ts:84]: These changes are not necessary (at least on my side) to get correct types in the provided sample code: https://github.com/bogeychan/elysia-logger/issues/14#issuecomment-2016045792

Can you please share some code that justifies the changes suggested here?

**Review by bogeychan — CHANGES_REQUESTED** [review]

File: src/types.ts (hunk 2 above)
Comment (MariNatsuki) [reply src/types.ts:84]: something like this, the new Elysis's generic is different for the derive type, so you will get a type error when you cast the type of context with InferContext in CustomProps or autoLogging

```
const app = new Elysia()
  .derive(() => ({
    foo: 'bar',
  }));

app
  .use(
    logger({
      customProps({ foo }: InferContext<typeof main>) {
        return {
          foo,
        };
      },
      autoLogging: true,
    })
  )
```

**Review by bogeychan — APPROVED** [review]
Thanks for your contribution!

i changed a few things and added tests

**Pull request merged**